packages/core/src/connectivity/scp-cf/token-accessor.ts

```diff
@@ -40,40 +69,48 @@ export async function serviceToken(
       userJwt?: string | JwtPayload;
     }
 ): Promise<string> {
-  const resolvedService = resolveService(service);
-
   const opts = {
     useCache: true,
     enableCircuitBreaker: true,
     ...options
   };
 
+  service = resolveService(service);
+  const serviceCredentials = service.credentials;
   const xsuaa = multiTenantXsuaaCredentials(opts.userJwt);
-  const serviceCreds = extractClientCredentials(resolvedService.credentials);
 
   if (opts.useCache) {
     const cachedToken = clientCredentialsTokenCache.getGrantTokenFromCache(
       xsuaa.url,
-      serviceCreds
+      serviceCredentials.clientid
     );
     if (cachedToken) {
       return cachedToken.access_token;
     }
   }
+
+  // TODO: handle Payload as well
+  let userJwt = options?.userJwt;
+  if (typeof userJwt !== 'string' && typeof userJwt !== 'undefined') {
+    userJwt = undefined;
+  }
+
   try {
-    const resp = await clientCredentialsGrant(xsuaa, serviceCreds, opts);
+    const token = await getClientCredentialsToken(service, userJwt);
+
     if (opts.useCache) {
       clientCredentialsTokenCache.cacheRetrievedToken(
         xsuaa.url,
-        serviceCreds,
-        resp
+        serviceCredentials.clientid,
+        token
       );
     }
-    return resp.access_token;
-  } catch (error) {
+    // TODO: Token responses can have id_token instead of access_token (xssec does that correctly)
+    return token.access_token;
+  } catch (err) {
     throw new ErrorWithCause(
-      `Fetching an access token for service "${resolvedService.label}" failed.`,
-      error
+      `Could not fetch client credentials token for service of type "${service.label}".`,
+      err
```

Comment:
I put the comment here but it is about the `jwtBearerToken` in this file. From the docs of xssec it is stated that also grant type `urn:ietf:params:oauth:grant-type:saml2-bearer` is supported so we could also replace the method call in there or deprecate the method and ask users to use the xssec lib. I guess the validation could be a potential issue, but for this grant I think it is fair to validate the JWT before even if it is a breaking change in behavior.

Reply:
Same as above. Agreed, but I would do it on a separate PR.

Reply:
ok